bridges/otelslog/handler_test.go

```diff
@@ -611,3 +612,57 @@ func BenchmarkHandler(b *testing.B) {
 
 	_, _ = h, err
 }
+
+func TestHandler_convertRecord(t *testing.T) {
+	// Capture the PC of this line
+	pc, file, line, _ := runtime.Caller(0)
+	funcName := runtime.FuncForPC(pc).Name()
+
+	tests := []struct {
+		name      string
+		handler   Handler
+		wantAttrs []log.KeyValue
+	}{
+		{
+			name:      "empty",
+			handler:   Handler{},
+			wantAttrs: []log.KeyValue{},
+		},
+		{
+			name:    "with source",
+			handler: Handler{source: true},
+			wantAttrs: []log.KeyValue{
+				{Key: "source", Value: log.MapValue(
+					log.String("function", funcName),
+					log.String("file", file),
+					log.Int("line", line),
+				)},
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			slogRecord := slog.NewRecord(time.Now(), slog.LevelInfo, "body", pc)
+			record := tt.handler.convertRecord(slogRecord)
+
+			// Validate attributes
+			attrMap := make(map[string]bool)
+			for _, attr := range tt.wantAttrs {
+				attrMap[attr.String()] = true
```

Comment:
Do not use [`KeyValue.String`](https://pkg.go.dev/go.opentelemetry.io/otel/log#KeyValue.String) for comparing. : 

> The returned string is meant for debugging; the string representation is not stable.

Use [`KeyValue.Equal`](https://pkg.go.dev/go.opentelemetry.io/otel/log#KeyValue.Equal) or [`Value.Equal`](https://pkg.go.dev/go.opentelemetry.io/otel/log#Value.Equal) for comparing.